packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -781,3 +800,110 @@ void _exitWithExpectedFileNotFound({
+      'Failed to create the symlink $linkPath->$targetPath: $exception'
+    );
+  }
+}
+
+String _getLocalArtifactVersion(String pomPath) {
+  final File pomFile = fs.file(pomPath);
+  if (!pomFile.existsSync()) {
+    throwToolExit('The file $pomPath wasn\'t found in the local engine out directory.');
+  }
+  xml.XmlDocument document;
+  try {
+    document = xml.parse(pomFile.readAsStringSync());
+  } on xml.XmlParserException {
+    throwToolExit(
+      'Error parsing $pomPath. Please ensure that this is a valid XML document.'
+    );
+  } on FileSystemException {
+    throwToolExit(
+      'Error reading $pomPath. Please ensure that you have read permission to this'
+      'file and try again.');
+  }
+  final Iterable<xml.XmlElement> project = document.findElements('project');
+  assert(project.isNotEmpty);
+  for (xml.XmlElement versionElement in document.findAllElements('version')) {
+    if (versionElement.parent == project.first) {
+      return versionElement.text;
+    }
+  }
+  throwToolExit('Error while parsing the <version> element from $pomPath');
+  return null;
+}
+
+/// Returns the local Maven repository for a local engine build.
+/// For example, if the engine is built locally at <home>/engine/src/out/android_release_unopt
+/// This method generates symlinks in the temp directory to the engine artifacts
+/// following the convention specified on https://maven.apache.org/pom.html#Repositories
+Directory _getLocalEngineRepo({
+  @required String engineOutPath,
+  @required AndroidBuildInfo androidBuildInfo,
+}) {
+  assert(engineOutPath != null);
+  assert(androidBuildInfo != null);
+
+  final String abi = getEnumName(androidBuildInfo.targetArchs.first);
+  final Directory localEngineRepo = fs.systemTempDirectory
+    .createTempSync('flutter_tool_local_engine_repo.');
+
+  // Remove the local engine repo before the tool exits.
+  addShutdownHook(
+    () => localEngineRepo.deleteSync(recursive: true),
+    ShutdownStage.CLEANUP,
+  );
+
+  final String buildMode = androidBuildInfo.buildInfo.modeName;
+  final String artifactVersion = _getLocalArtifactVersion(
+    fs.path.join(
+      engineOutPath,
+      'flutter_embedding_$buildMode.pom',
+    )
+  );
+  for (String artifact in const <String>['pom', 'jar']) {
+    // The Android embedding artifacts.
+    _createSymlink(
+      fs.path.join(
+        engineOutPath,
+        'flutter_embedding_$buildMode.$artifact',
+      ),
+      fs.path.join(
+        localEngineRepo.path,
+        'io',
+        'flutter',
+        'flutter_embedding_$buildMode',
+        artifactVersion,
+        'flutter_embedding_$buildMode-$artifactVersion.$artifact',
+      ),
+    );
+    // The engine artifacts (libflutter.so).
+    _createSymlink(
+      fs.path.join(
+        engineOutPath,
+        '${abi}_$buildMode.$artifact',
+      ),
+      fs.path.join(
+        localEngineRepo.path,
+        'io',
+        'flutter',
+        '${abi}_$buildMode',
+        artifactVersion,
```

Comment:
Why not just have the engine do this when it builds the artifacts?

I'm trying to build an add-to-app case here and have to reproduce this manually since I can't use the tool to do it.

Reply:
Have you tried to build an AAR and then make the host app depend on the AAR?

Reply:
Yes - but the host app still doesn't get the right local engine assets unless I run `gradlew -Plocal-engine-repo` etc on it - and I don't get this repo created when I build the aar with `--local-engine`.